plugins/out_es/es.h

```diff
@@ -66,14 +70,17 @@ struct flb_elasticsearch {
     char *aws_sts_endpoint;
     char *aws_profile;
     struct flb_aws_provider *aws_provider;
+    int own_aws_provider;
```

Comment:
We shouldn't use this pattern. This is because when initializing struct typed values or basic type of values, we can distinguish whether allocated or not.

Reply:
Answered in [another comment](https://github.com/fluent/fluent-bit/pull/7608#issuecomment-2526291616). `own_xxx` fields are used to deallocate data shared b/w plugin configuration and upstream node configuration just once.